compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/DeclarationsConverter.kt

```diff
@@ -379,7 +379,7 @@ class DeclarationsConverter(
                     isInner = modifiers.isInner()
                     isCompanion = modifiers.isCompanion() && classKind == ClassKind.OBJECT
                     isData = modifiers.isDataClass() && classKind != ClassKind.OBJECT
-                    isInline = modifiers.hasInline()
+                    isInline = modifiers.isInlineClass() && classKind != ClassKind.OBJECT
```

Comment:
I wonder, though, that both `isData` and `isInline` should check `classKind == ClassKind.CLASS` instead?

Reply:
This may be a silly question, but why is the `classKind` check necessary at all? The old frontend just [looks at the modifier list](https://github.com/JetBrains/kotlin/blob/1a26d2e157a31459d9bf3abca0123f03a6ae011d/compiler/frontend/src/org/jetbrains/kotlin/resolve/lazy/descriptors/LazyClassDescriptor.java#L156) and psi2ir sets the isInline flag [directly based on the descriptor](https://github.com/JetBrains/kotlin/blob/1a26d2e157a31459d9bf3abca0123f03a6ae011d/compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/declarations/impl/IrClassImpl.kt#L74).

Reply:
Not a silly question, thanks for the pointers! @mglukhikh any reasons we check that? I saw some diagnostic tests check invalid uses of that modifier, though.

Hm, I just checked the grammar: https://kotlinlang.org/docs/reference/grammar.html#classModifier and `inline` is not a class modifier?!

Reply:
I will think about this, thanks.

Reply:
Probably these checks are not needed, I will try to remove them.